legacyman_parser/parse_class_attributes_from_tonals.py

```diff
@@ -21,15 +21,22 @@
 
 def extract_class_attributes_from_tonals_page(soup: BeautifulSoup = None, parsed_url: str = None,
                                               parent_url: str = None, userland_dict: dict = None) -> []:
-    propulsion_h1 = soup.find('h1')
-    assert "PROPULSION" in propulsion_h1.text.upper(), "InvalidAssumption: First H1 is going to be Propulsion in " \
-                                                       "tonals page {}".format(parsed_url)
-    propulsion_system_table_data = propulsion_h1.find_next('table').find_all('td')
-    assert propulsion_system_table_data[0]['colspan'] == '4', "InvalidAssumption: First row in propulsion table " \
-                                                              "is going to be unit title : {}".format(parsed_url)
-    assert len(propulsion_system_table_data) % 2 == 1, "InvalidAssumption: Propulsion table is in" \
-                                                       " Key, Value format, with header : {}".format(parsed_url)
-
+    # Find all tables on page
+    all_tables = soup.find_all('table')
+    propulsion_system_table = None
```

Comment:
What? You're not using `filter` functions?, like the ones I only learned about this week?